### PR TITLE
Settings UI: add translation comment.

### LINKED ISSUE #6935: i18n: strings need translator comments
The following strings are difficult to translate without some context:

https://github.com/Automattic/jetpack/blob/master/_inc/client/traffic/related-posts.jsx#L118
https://github.com/Automattic/jetpack/blob/master/_inc/client/traffic/related-posts.jsx#L123
https://github.com/Automattic/jetpack/blob/master/_inc/client/traffic/related-posts.jsx#L128

After reading the code I understand these are parts of sample related posts, specifically categories? Can a translator comment please be added to clarify that?

### PR DESCRIPTION
**Pull request by eliorivero**
Fixes #6935

#### Changes proposed in this Pull Request:

- add translation comment
- remove word that wasn't supposed to be there
- Fix minor ESLint warnings.

### CODE REVIEW

File: _inc/client/traffic/related-posts.jsx [hunk 1]
```diff
@@ -113,19 +113,25 @@ export const RelatedPosts = moduleSettingsForm(
 								{
 									[
 										{
-											url : '1-wpios-ipad-3-1-viewsite.png',
+											url: '1-wpios-ipad-3-1-viewsite.png',
 											text: __( 'Big iPhone/iPad Update Now Available' ),
-											context: __( 'In "Mobile"' )
+											context: __( 'In "Mobile"', {
+												comment: 'It refers to the category where a post was found. Used in an example preview.'
```
Comment (zinigor) [inline _inc/client/traffic/related-posts.jsx]: It should be `context`, not `comment` here and in the other two instances.

Comment (eliorivero) [inline _inc/client/traffic/related-posts.jsx]: I got this from
https://github.com/Automattic/i18n-calypso#more-translate-examples
specifically this one:
```js
// add a comment to the translator
var content = i18n.translate( 'g:i:s a', {
        comment: 'draft saved date format, see http://php.net/date'
    } );
```

**Comment by eliorivero** [comment]
Updated to use `context` instead of `comment`in i18n `translate` function.

**Comment by yoavf** [comment]
Sorry to intrude, but @zinigor - why context?
Context needs to be used in very specific places. For example, do differentiate between "Post" (verb), and "Post" (noun)

In most other cases, like this one, a translator comment should be used instead.

When you add a context to a string, you are creating a new string to be translated, while a context keeps the existing translations, but just adds clarification.

**Comment by zinigor** [comment]
Oh, thanks for clarifying @yoavf ! My initial idea was that you need to use context to describe the phrase's surroundings and it's, well, usage context. While a translator comment was needed to describe things used inside of the string like placeholders, etc.

These strings are new, in that case what would you suggest us use, `context` or `comment'?

**Comment by eliorivero** [comment]
@zinigor I should have checked the code before changing it after our Slack exchange.

From what can be seen here, 
https://github.com/Automattic/i18n-calypso/blob/d56d4929bdc2606862f256660febcf6599e44252/cli/formatters/php.js#L41-L49

the effect of `context` is that of a `_x` while `comment` pushes an `/**/` comment.

I've changed them back to `comment`.

**Comment by yoavf** [comment]
> My initial idea was that you need to use context to describe the phrase's surroundings and it's, well, usage context. 
It's often used that way, but it should be more specific. Basically, if you're adding a new string - you probably don't need a context.
If you add a string that exists, but with another usage, that's when you want a context.
There are exceptions of course, but that's a good rule to follow :)

**Comment by dereksmart** [comment]
pushed to 4.8 0cf17db2d

**Pull request merged**